core/src/main/java/org/infinispan/remoting/transport/jgroups/EmbeddedJGroupsChannelConfigurator.java

```diff
@@ -128,12 +132,25 @@ public JChannel createChannel() throws Exception {
          }
       }
 
-
       if (!hasRelay2 && hasSites()) {
          throw CONFIG.jgroupsRemoteSitesWithoutRelay(name);
       }
 
-      return amendChannel(new JChannel(protocols));
+      // Need to initialize components including injecting component attributes (ie. TP.msg_processing_policy.max_buffer_size)
```

Comment:
I'm not sure why this is needed. `JChannel` already invokes `init()` and `initComponents()`

Reply:
I really don't like to invoke `init()` twice, but this can be simplified as
```java
      JChannel channel = new JChannel(protocols);
      channel.getProtocolStack().initProtocolStack(actualStack);
      return amendChannel(channel);
```

Reply:
> I'm not sure why this is needed. `JChannel` already invokes `init()` and `initComponents()`

The problem isn't the init itself, it is the attributes replacement. We call `Configurator.initializeAttrs(protocol, configuration, stackType);` but that only replaces the attributes that don't have components.

> I really don't like to invoke `init()` twice, but this can be simplified as
> 
> ```java
>       JChannel channel = new JChannel(protocols);
>       channel.getProtocolStack().initProtocolStack(actualStack);
>       return amendChannel(channel);
> ```

I don't like invoking init twice either, but there is no way that I found to replace the component attributes until after the JChannel is created. I do like the code you posted better though.